### PR TITLE
[fix](catalog) disable FileSystem Cache to avoid too many fs cache

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

When creating a new hive catalog or refresh the hive catalog, it will refresh the HiveMetaStore cache.
And it will call "FileInputFormat.setInputPaths()".
In this method, it will create a new FileSystem instance and store it in FileSystem's cache.
So if refresh catalog frequently, there will be too many FileSystem instances in cache, causing OOM.

This PR disable the FileSystem Cache.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java [hunk 1]
```diff
@@ -328,6 +329,17 @@ private FileCacheValue loadFiles(FileCacheKey key) {
         try {
             Thread.currentThread().setContextClassLoader(ClassLoader.getSystemClassLoader());
             String finalLocation = S3Util.convertToS3IfNecessary(key.location);
+            // disable the fs cache in FileSystem, or it will always from new FileSystem
+            // and save it in cache when calling FileInputFormat.setInputPaths().
+            try {
+                Path path = new Path(finalLocation);
+                URI uri = path.toUri();
+                if (uri.getScheme() != null) {
+                    updateJobConf("fs." + uri.getScheme() + ".impl.disable.cache", "true");
```
Comment (wsjz) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java:338]: s3 scheme: s3 and s3a
cos scheme: cosn

**Review by wsjz — APPROVED** [review]

**Review by wsjz — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java (hunk 1 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java:338]: It does need to be same logic as in `FileInputFormat.setInputPaths()`, which is to use `schema` is file location.

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.84 seconds
 stream load tsv:          451 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          70.1 seconds inserted 10000000 Rows, about 142K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230628104032_clickbench_pr_169312.html

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**